features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/TestConsole/TestConsole.jsx

```diff
@@ -134,121 +162,81 @@ class TestConsole extends React.Component {
-                }
-                if (apiData.scopes) {
-                    const scopeList = apiData.scopes.map((scope) => { return scope.name; });
-                    this.setState({ scopes: scopeList });
+                environments = apiResponse.body.map((env) => { return env.name; });
+                if (environments && environments.length > 0) {
+                    [selectedEnvironment] = environments;
+                    this.setState({
+                        environments,
+                        selectedEnvironment,
+
+                    });
                 }
+                return Api.getAPIById(apiObj.id);
+            })
+            .then((apiResponse) => {
+                apiData = apiResponse.obj;
+                this.setState({
+                    api: apiData,
+                });
                 if (environments && environments.length > 0) {
                     [selectedEnvironment] = environments;
-                    return API.getSwaggerByAPIIdAndEnvironment(apiResponse.obj.id, selectedEnvironment);
-                } else if (labels && labels.length > 0) {
-                    [selectedEnvironment] = labels;
-                    return API.getSwaggerByAPIIdAndLabel(apiResponse.obj.id, selectedEnvironment);
+                    return Api.getSwaggerByAPIIdAndEnvironment(apiResponse.obj.id, selectedEnvironment);
                 } else {
-                    return API.getSwaggerByAPIId(apiResponse.obj.id);
+                    return Api.getSwaggerByAPIId(apiResponse.obj.id);
                 }
             })
             .then((swaggerResponse) => {
                 swagger = swaggerResponse.obj;
-                if (user != null) {
-                    this.setState({
-                        api: apiData,
-                        swagger,
-                        environments,
-                        labels,
-                        productionAccessToken,
-                        sandboxAccessToken,
-                    });
-                    return API.getSubscriptions(apiID);
-                } else {
-                    return null;
-                }
+                this.setState({
+                    swagger,
+                });
+                return Api.getSettings();
             })
-            .catch((error) => {
-                if (process.env.NODE_ENV !== 'production') {
-                    Alert.error(error);
-                    console.error(error);
-                }
-                this.setState({ serverError: `${error.statusCode} - ${error.response.body.description}` });
-            });
-        const settingPromise = API.getSettings();
-        settingPromise
-            .then((settingsNew) => {
-                if (settingsNew.environment) {
-                    urls = settingsNew.environment.map((environment) => {
+            .then((settingsObj) => {
+                if (settingsObj.environment) {
+                    urls = settingsObj.environment.map((envo) => {
```

Comment:
This `urls ` seems to have declared outside the scope of then callback, so assigning a value to it and using it outside the then callback scope could lead to unexpected results, we needs to get the variable inside the callback method